src/proxy.ts

```diff
@@ -32,11 +32,15 @@ export async function proxyRequest(url: string, context: Context, opts: ProxyOpt
   const start = Date.now()
 
   // fetch response from upstream app
-  context.response = await fetch(context.proxy)
+  try {
+    context.response = await fetch(context.proxy);
+  } catch (err) {
+    context.response = new Response(err.message, { status: 502 });
```

Comment:
I'm OK with this as a https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/502 communicates there was an issue. It would also be nice to differentiate for when a timeout occurred which would then be a https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/504.

Reply:
I agree. I'll open another PR with timeouts, which has some other implications